### PR TITLE
add Push UPS template

### PR DESCRIPTION
**Pull request by philbrookes**
to test:
- Deploy docker.io/feedhenry/negotiator:0.0.8 
- have data-mysql running in the environment
- execute this curl:
```bash
curl -X POST http://<negotiator>/service/deploy/push-ups/<env> -d '{"serviceName":"push-ups","domain":"<domain>", "replicas":1, "target":{"host":"<target>,"token":"<token>"}}'
```
- See UPS deploy successfully.

### CODE REVIEW

File: pkg/deploy/configuration.go [hunk 1]
```diff
@@ -544,3 +566,20 @@ func (d *DataMysqlConfigure) Configure(client Client, deployment *dc.DeploymentC
 
 	return deployment, nil
 }
+
+func (p *PushUpsConfigure) statusUpdate(key, message, status string) {
+	if p.status == nil {
```
Comment (maleck13) [inline pkg/deploy/configuration.go]: Just so you know there is a change to the status api in the other PR here

File: pkg/openshift/templates/push-ups.json.tpl.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    	{
+			"kind": "Service",
+			"apiVersion": "v1",
+			"metadata": {
+				"name": "{{.ServiceName}}",
+				"labels": {
+					"name": "{{.ServiceName}}",
+					"rhmap/domain": "{{.Domain}}",
+					"rhmap/env": "{{.Env}}",
+					"rhmap/guid": "{{.CloudAppGUID}}",
```
Comment (maleck13) [inline pkg/openshift/templates/push-ups.json.tpl.go]: Having this label on everything has been causing some problems where we use the guid if it is present but most of the time it is an empty string. I think we should remove it from all but the cloud app. or else generate a unique value for it. Maybe create an issue for that. No need to change right now

**Comment by maleck13** [comment]
I accidentally push 0.0.8 I have bumped to 0.0.9 now but you need to repush

File: pkg/openshift/templates/push-ups.json.tpl.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+var PushUPSTemplate = `
+{{define "push-ups"}}
+{
+  	"kind": "Template",
+  	"apiVersion": "v1",
+  	"metadata": {
+    	"name": "RHMAP UPS",
+    	"annotations": {
+      		"description": "RHMAP ups",
+			"dependencies": "data-mysql",
```
Comment (philbrookes) [inline pkg/openshift/templates/push-ups.json.tpl.go]: at the moment I'm using space-seperated in that work, would you prefer comma-seperated? I don't really have a strong opinion either way.

File: pkg/openshift/templates/push-ups.json.tpl.go (hunk 2 above)
Comment (philbrookes) [inline pkg/openshift/templates/push-ups.json.tpl.go]: Yeah, I actually changed it, in my dependency work to:
```go
"rhmap/guid": "{{ genPass }}",
```

**Pull request merged**